analyzer/tests/analysis.rs

```diff
@@ -1,105 +1,181 @@
-#![cfg(feature = "ignore")]
 use fe_analyzer;
-use fe_analyzer::namespace::types::{Array, Base, Map, Type};
-use fe_analyzer::{ExpressionAttributes, Location};
-use fe_parser::ast as fe;
-use fe_parser::node::{Node, Span};
+use fe_analyzer::context::Context;
+use fe_common::diagnostics::{diagnostics_string, Diagnostic, Label};
+use fe_common::files::{FileStore, SourceFileId};
+use fe_parser::node::Span;
+use insta::assert_snapshot;
+use rstest::rstest;
+use std::collections::hash_map::DefaultHasher;
+use std::fmt::Debug;
+use std::fs;
+use std::hash::{Hash, Hasher};
 
-const GUEST_BOOK: &str = include_str!("fixtures/guest_book.fe");
+#[rstest(
+    fixture,
+    case::erc20_token("demos/erc20_token.fe"),
+    case::guest_book("demos/guest_book.fe"),
+    case::uniswap("demos/uniswap.fe"),
+    case::address_bytes10_map("features/address_bytes10_map.fe"),
+    case::assert("features/assert.fe"),
+    case::aug_assign("features/aug_assign.fe"),
+    case::base_tuple("features/base_tuple.fe"),
+    case::call_statement_with_args("features/call_statement_with_args.fe"),
+    case::call_statement_with_args_2("features/call_statement_with_args_2.fe"),
+    case::call_statement_without_args("features/call_statement_without_args.fe"),
+    case::checked_arithmetic("features/checked_arithmetic.fe"),
+    case::constructor("features/constructor.fe"),
+    case::create2_contract("features/create2_contract.fe"),
+    case::create_contract("features/create_contract.fe"),
+    case::create_contract_from_init("features/create_contract_from_init.fe"),
+    case::empty("features/empty.fe"),
+    case::events("features/events.fe"),
+    case::external_contract("features/external_contract.fe"),
+    case::for_loop_with_break("features/for_loop_with_break.fe"),
```

Comment:
Just to double check: We now test the state of the context for all these fixtures as an additional category of tests that we run. These tests do not actually test the return values of the called functions which we continue to do as before here. Is that right?

https://github.com/ethereum/fe/blob/c3b1f31f5dd01d6bb108beb856c905654771b4e2/compiler/tests/cases/features.rs#L112-L237

Reply:
Yes, we are just testing the state of the context. Anytime we change something in the analyzer that results in a different context output for a given fixture, it will show up here.

There have been a number of times where I've been reworking things in the analyzer and ended up breaking some unrelated full-compilation fixture test. Resolving these types of issues is tedious, since it's not clear what changed in the analyzer output. With this, it will be obvious what changed.